mantidproject/mantid
# Bug superplot tight layout

**Description of work.**

This is a continuation of #34885.

I realized that the fix provided by #34885 only applies to plot updates. But the bug is still present when the plot is freshly opened on data with a long name.

Because the issue was related to the plot legend, I took advantage of this PR to set the legend visibility according to the plot context menu option.

**To test:**

To reproduce the issue:
* create a workspace with a long name
```python
from mantid.simpleapi import CreateSampleWorkspace
CreateSampleWorkspace(OutputWorkspace="verrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrrryLonnnnnnnnnnnnnnnnnnnnnnnnnnngNaaaaaaaaaaaaaaaaaaaaaaaame")
```
* and open the superplot widget on it (right click -> plot -> superplot)
* the plot area should be displayed properly

And check that the legend visibility respects the checkbox state in the right click menu of the plot.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
